src/lib/flow/sol-flow-js.c

```diff
@@ -0,0 +1,1422 @@
+            return false;
+        }
+
+        /* This is in order to get port methods references in one call.
+         *
+         * We have 2 methods for each output port. We put all in the stash,
+         * even with 'undefined' values, if the method is not implemented on JS.
+         *
+         * We calculate the index by the following:
+         *
+         * base + output_port_index * ports_out_methods_length + method_index
+         *
+         * base - where should it start, for output ports it should be the size of input ports * number of input methods.
+         * input_port_index - the index of the JS 'in' array entry.
+         * method_index - the index of the method for output ports.
+         */
+
+        duk_get_prop_string(duk_ctx, -1, "connect");
+        duk_put_prop_index(duk_ctx, -3, base + i * PORTS_OUT_METHODS_LENGTH + PORTS_OUT_CONNECT_INDEX);
+
+        duk_get_prop_string(duk_ctx, -1, "disconnect");
+        duk_put_prop_index(duk_ctx, -3, base + i * PORTS_OUT_METHODS_LENGTH + PORTS_OUT_DISCONNECT_INDEX);
+
+        duk_pop(duk_ctx); /* array entry */
+    }
+
+    duk_pop_2(duk_ctx); /* out array and global_stash value */
+
+    return true;
+}
+
+static bool
+setup_ports_methods(duk_context *duk_ctx, uint16_t ports_in_len, uint16_t ports_out_len)
+{
+    /* We're using duktape global stash to keep reference to some JS
+     * port methods: connect(), disconnect() and process() in order
+     * to call it directly when receive a port number.
+     */
+
+    if (!setup_ports_in_methods(duk_ctx, ports_in_len, 0))
+        return false;
+
+    if (!setup_ports_out_methods(duk_ctx, ports_out_len, ports_in_len * PORTS_IN_METHODS_LENGTH))
+        return false;
+
+    return true;
+}
+
+static int
+flow_js_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    const struct flow_js_type *type = (struct flow_js_type *)node->type;
+    struct flow_js_data *mdata = data;
+
+    mdata->duk_ctx = duk_create_heap_default();
+    if (!mdata->duk_ctx) {
+        SOL_ERR("Failed to create a Duktape heap");
+        return -1;
+    }
+
+    /* TODO: Check if there's a "already parsed" representation that we can use. */
+    if (duk_peval_lstring(mdata->duk_ctx, type->js_content_buf, type->js_content_buf_len) != 0) {
+        SOL_ERR("Failed to read from javascript content buffer: %s", duk_safe_to_string(mdata->duk_ctx, -1));
+        duk_destroy_heap(mdata->duk_ctx);
+        return -1;
+    }
+    duk_pop(mdata->duk_ctx); /* duk_peval_lstring() result */
+
+    duk_push_global_object(mdata->duk_ctx);
+
+    /* "Soletta_node_pointer" is a hidden property. \xFF is used to give one extra level of hiding */
+    duk_push_string(mdata->duk_ctx, "\xFF" "Soletta_node_pointer");
+    duk_push_pointer(mdata->duk_ctx, node);
+    duk_def_prop(mdata->duk_ctx, -3, DUK_DEFPROP_HAVE_VALUE |
+        DUK_DEFPROP_HAVE_WRITABLE | 0 |
```

Comment:
I used the way it is on duktape documentation. I spoke with guys on duktape channel and they told me that it makes sense at the end and not in between, in order to easily comment the arguments. I'll drop the '0's, thanks for pointing this out @lpereira